controllers/networkplugins_controller.go

```diff
@@ -17,35 +17,584 @@ limitations under the License.
+		whereaboutsConfig := (*WhereaboutsT)(old.Whereabouts)
+		err := r.deletePlugin(whereaboutsConfig)
+		if err != nil {
+			return err
+		}
+		*fileList = append(*fileList, "whereabouts.yaml")
+	}
+
+	if (noCni == true || r.currentSpec.CniPlugins.Sriov == nil) && old.Sriov != nil {
+		sriovConfig := (*SriovT)(old.Sriov)
+		err := r.deletePlugin(sriovConfig)
+		if err != nil {
+			return err
+		}
+		*fileList = append(*fileList, "sriov-cni.yaml")
+		*fileList = append(*fileList, "sriov-deviceplugin.yaml")
+	}
+
+	if (noCni == true || r.currentSpec.CniPlugins.HostPlumber == nil) && old.HostPlumber != nil {
+		hostPlumberConfig := (*HostPlumberT)(old.HostPlumber)
+		err := r.deletePlugin(hostPlumberConfig)
+		if err != nil {
+			return err
+		}
+		*fileList = append(*fileList, "hostplumber.yaml")
+	}
+
+	if (noCni == true || r.currentSpec.CniPlugins.NodeFeatureDiscovery == nil) && old.NodeFeatureDiscovery != nil {
+		nfdConfig := (*NodeFeatureDiscoveryT)(old.NodeFeatureDiscovery)
+		err := r.deletePlugin(nfdConfig)
+		if err != nil {
+			return err
+		}
+		*fileList = append(*fileList, "nfd.yaml")
+	}
+
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) createOrDeletePlugins(networkPlugins *plumberv1.NetworkPlugins, fileList []string, mode string) error {
+	for _, file := range fileList {
+		var fullpath string
+
+		switch mode {
+		case "create":
+			fullpath = filepath.Join(CreateDir, file)
+		case "delete":
+			fullpath = filepath.Join(DeleteDir, file)
+		default:
+			fmt.Printf("Invalid mode to applyDeleteTemplate: %s\n", mode)
+			return errors.NewBadRequest("Invalid option to applyDeleteTemplate")
+		}
+
+		fmt.Printf("mode = %s YAML file: %s\n", mode, fullpath)
+		yamlfile, err := ioutil.ReadFile(fullpath)
+
+		if err != nil {
+			fmt.Printf("Failed to read file: %s\n", fullpath)
+			return err
+		}
+
+		resource_list := []*unstructured.Unstructured{}
+		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(yamlfile), 4096)
+		for {
+			resource := unstructured.Unstructured{}
+			err := decoder.Decode(&resource)
+			if err == nil {
+				resource_list = append(resource_list, &resource)
+			} else if err == io.EOF {
+				break
+			} else {
+				fmt.Printf("Error decoding!!!: %s\n", err)
+				return err
+			}
+		}
+
+		for _, obj := range resource_list {
+			if mode == "create" {
+				fmt.Printf("Creating obj: %+v\n", obj)
+				err := apply.ApplyObject(context.Background(), r.Client, obj)
+				if err != nil {
+					fmt.Printf("Error creating object: %+v\n\nError: %s\n", obj, err)
+					return err
+				}
+			} else if mode == "delete" {
+				fmt.Printf("Deleting obj: %+v\n", obj)
+				err := apply.DeleteObject(context.Background(), r.Client, obj)
+				if err != nil {
+					fmt.Printf("Error deleting object: %+v\n\nError: %s\n", obj, err)
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (r *NetworkPluginsReconciler) saveSpecConfig(ctx context.Context, req ctrl.Request) error {
```

Comment:
See the earlier comment about using the `status` instead of storing the prev state in an external format/configMap.